onnxruntime/python/tools/transformers/convert_generation.py

```diff
@@ -1295,8 +1295,8 @@ def replace_mha_with_gqa(model: OnnxModel, past_seq_len_input: str, kv_num_heads
                 outputs=node.output,
                 name=node.name.replace("MultiHeadAttention", "GroupQueryAttention"),
                 domain="com.microsoft",
-                num_heads=node.attribute[0].i,
-                kv_num_heads=node.attribute[0].i if kv_num_heads == 0 else kv_num_heads,
+                num_heads=node.attribute[0].i // world_size,
```

Comment:
MHA has multiple attributes (mask_filter_value, num_heads, scale), and node.attribute[0] might or might not be num_heads. I suggest to use attribute name to look up the value.

Reply:
Good suggestion, will do